src/modding/ModHelpers.cs

```diff
@@ -0,0 +1,166 @@
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name) + "\n";
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_DEPENDENCIES_ORDER_FIX"),
+                    offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name);
+                break;
+            case (int)ModLoader.CheckErrorStatus.IncompatibleMod:
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_INCOMPATIBLE_MOD"), offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name) + "\n";
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_INCOMPATIBLE_MOD_FIX"),
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name);
+                break;
+            case (int)ModLoader.CheckErrorStatus.InvalidLoadOrderBefore:
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_LOAD_ORDER_BEFORE"), offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name) + "\n";
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_LOAD_ORDER_BEFORE_FIX"),
+                    offendingMod.Name,
+                    string.IsNullOrWhiteSpace(otherMod.Name) ? otherMod.InternalName : otherMod.Name);
+                break;
+            case (int)ModLoader.CheckErrorStatus.InvalidLoadOrderAfter:
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_LOAD_ORDER_AFTER"), offendingMod.Name,
+                    otherMod.Name) + "\n";
+                result += string.Format(TranslationServer.Translate("MOD_ERROR_LOAD_ORDER_AFTER_FIX"),
+                    offendingMod.Name, otherMod.Name);
+                break;
+        }
+
+        return result;
+    }
+
+    public static VersionCompatibility GetVersionCompatibility(ModInfo info)
+    {
+        var isCompatibleVersion = 0;
+        var compatibleVersionTest = 0;
+
+        if (!string.IsNullOrEmpty(info.MinimumThriveVersion) &&
+            VersionUtils.Compare(Constants.Version, info.MinimumThriveVersion ?? string.Empty) >= 0)
+        {
+            ++compatibleVersionTest;
+        }
+        else if (!string.IsNullOrEmpty(info.MinimumThriveVersion))
+        {
+            compatibleVersionTest--;
+        }
+
+        if (!string.IsNullOrEmpty(info.MaximumThriveVersion) &&
+            VersionUtils.Compare(Constants.Version, info.MaximumThriveVersion ?? string.Empty) >= 0)
+        {
+            ++isCompatibleVersion;
+        }
+        else if (!string.IsNullOrEmpty(info.MaximumThriveVersion))
+        {
+            compatibleVersionTest--;
+        }
+
+        if (compatibleVersionTest >= 1)
+        {
+            isCompatibleVersion = 1;
```

Comment:
I think it's more clearer if the enum is directly returned instead of typecasting it afterwards.